cpp/src/arrow/dataset/file_base.cc

```diff
@@ -89,6 +89,28 @@ Result<std::shared_ptr<io::InputStream>> FileSource::OpenCompressed(
   return io::CompressedInputStream::Make(codec.get(), std::move(file));
 }
 
+Result<std::shared_ptr<io::InputStream>> FileSource::OpenRange(int64_t start,
+                                                               int64_t end) const {
+  ARROW_ASSIGN_OR_RAISE(auto file, Open());
+
+  auto actual_compression = Compression::type::UNCOMPRESSED;
+
+  auto extension = fs::internal::GetAbstractPathExtension(path());
+  if (extension == "gz") {
```

Comment:
I'm curious, why is this special case needed?

Reply:
The CSV reader does this in other places probably to save time and not run the getcompressionType. So I just copy it here.